Automattic/jetpack
# Premium Content Block: restrict registration to wpcom

Fixes #18521

#### Changes proposed in this Pull Request:

We already do the same on the PHP side; let's also limit the use of the block to WordPress.com sites (simple and atomic).

**Note**: by doing so, we would break the block for anyone using it on their self-hosted site today by using the experimental or beta flags. However, I should note that we intended for the block to only be available on WordPress.com from the start (see d80d3d96cd3c762007e5f67b4a802c955a796621 and c39578af43b89097c53f38ff79d62ce960775321 in the PR that introduced the feature).

#### Jetpack product discussion

* See https://github.com/Automattic/jetpack/issues/18521#issuecomment-1031659152

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* On a Jetpack site that's connected to WordPress.com, and with a complete plan, enable Beta blocks: `define( 'JETPACK_BETA_BLOCKS', true );` in `wp-config.php`.
* Go to Posts > Add New
* Search for the Premium Content block; it should not be offered to you.
* On a WoA site (or a wpcom simple site with a paid plan running D74536-code), the block should still be available.
